marklogic-data-hub/src/main/java/com/marklogic/hub/hubcentral/migration/HubCentralMigrator.java

```diff
@@ -0,0 +1,166 @@
+                List<String> pathRangeIndex = mapper.convertValue(entityTypeNode.get("pathRangeIndex"), ArrayList.class);
+                pathRangeIndex = pathRangeIndex == null ? new ArrayList<>() : pathRangeIndex;
+                Set<String> mergedIndexArrays = new HashSet<>();
+                Stream.of(elementRangeIndex, rangeIndex, pathRangeIndex).forEach(mergedIndexArrays::addAll);
+
+                ObjectNode entityTypePropertiesNode = (ObjectNode) entityTypeNode.get("properties");
+                if (entityTypePropertiesNode == null) {
+                    logger.warn("entityTypePropertiesNode is null");
+                    entityTypeNode.remove(removableIndexArrays);
+                    atLeastOneEntityModelWasMigrated = true;
+                    continue;
+                }
+
+                entityTypePropertiesNode.fieldNames().forEachRemaining(propertyName -> {
+                    if (mergedIndexArrays.contains(propertyName)) {
+                        ObjectNode entityPropertyNode = (ObjectNode) entityTypePropertiesNode.get(propertyName);
+                        if (!isStructuredTypeProperty(entityPropertyNode)) {
+                            entityPropertyNode.put("facetable", true);
+                        }
+                    }
+                });
+                entityTypeNode.remove(removableIndexArrays);
+
+                File entityModelFile = Paths.get(hubProject.getHubEntitiesDir().toString(), title + entityManager.ENTITY_FILE_EXTENSION).toFile();
+                try {
+                    writer.writeValue(entityModelFile, entityModelNode);
+                    logger.warn(format("Entity Model '%s' was successfully migrated", entityModelFile));
+                    atLeastOneEntityModelWasMigrated = true;
+                } catch (IOException e) {
+                    logger.error(format("Entity Model '%s' migration failed; cause: %s", entityModelFile, e.getMessage()), e);
+                }
+            }
+        }
+
+        if (atLeastOneEntityModelWasMigrated) {
+            logger.warn("Finished migrating entity models.");
+            logger.warn("Please examine your entity model files to verify that properties that were listed in the rangeIndex, pathRangeIndex, or elementRangeIndex arrays " +
+                    "now have \"facetable\":true in their property definition.\n");
+        } else {
+            logger.warn("No entity models required migration, so no project files were modified");
+        }
+    }
+
+    protected boolean entityModelRequiresMigration(ObjectNode entityModelNode) {
+        if (!entityModelValidForMigration(entityModelNode)) {
+            return false;
+        }
+        String firstLevelEntityTypeName = entityModelNode.get("info").get("title").asText();
+        entityModelNode = (ObjectNode) entityModelNode.get("definitions").get(firstLevelEntityTypeName);
+        return entityModelNode.get("rangeIndex") != null || entityModelNode.get("elementRangeIndex") != null ||
```

Comment:
Going to be a little picky here - I think it's worth verifying that for each property, it both exists and the array has size of 1 or more. That's because I think it's common for entity definitions to be stubbed out with empty arrays. No need to process those.

Reply:
1. All the entity properties are iterated over to see if an entityPropertyName exist in one of the index arrays. So if something which is in an index array and not defined in entityTypeProperties will not cause an issue and the index arrays are deleted after adding facetable properties.
1. If the array is empty (size is 0), we delete those arrays on migration. (Since we dont want those references existing after migration)